react-app/src/hooks/api/useBuildingsApi.ts

```diff
@@ -1,55 +1,104 @@
-const buildings1 = [
-  {
-    Id: 1,
-    ClassificationId: 0,
-    Classification: {
-      Name: 'Core operational',
-      Id: 0,
-    },
-    Address1: '2345 Example St.',
-    AgencyId: 1,
-    Agency: { Name: 'Smith & Weston' },
-    PID: 111333444,
-    IsSensitive: true,
-    UpdatedOn: new Date(),
-    Evaluations: [{ Value: 99888, Date: new Date() }],
-    Fiscals: [{ Value: 1235000, FiscalYear: 2024 }],
-  },
-  {
-    Id: 2,
-    ClassificationId: 5,
-    Classification: {
-      Name: 'Disposed',
-      Id: 5,
-    },
-    Address1: '6432 Nullabel Ln.',
-    AgencyId: 1,
-    Agency: { Name: 'Smith & Weston' },
-    PID: 676555444,
-    IsSensitive: false,
-    UpdatedOn: new Date(),
-    Evaluations: [{ Value: 999988, Date: new Date() }],
-    Fiscals: [{ Value: 1235000, FiscalYear: 2024 }],
-  },
-];
+import { IFetch } from '../useFetch';
+import { Agency } from './useAgencyApi';
+export interface Evaluation {
+  Date: Date;
+  Value: number;
+}
+export interface Fiscal {
+  FiscalYear: number;
+  EffectiveDate: Date;
+  Value: number;
+}
+export interface Classification {
+  Name: string;
+}
+export interface BuildingConstructionType {
+  Name: string;
+}
+export interface BuildingPredominateUse {
+  Name: string;
+}
+export interface BuildingOccupantType {
+  Name: string;
+}
+export interface PropertyType {
+  Name: string;
+}
 
-const useBuildingsApi = () => {
+export type GeoPoint = {
+  x: number;
+  y: number;
+};
+export interface Building {
+  Id?: string;
+  Name?: string;
+  Description?: string;
+  BuildingConstructionTypeId: number;
+  BuildingConstructionType?: BuildingConstructionType;
+  BuildingFloorCount: number;
+  BuildingPredominateUseId: number;
+  BuildingPredominateUse: BuildingPredominateUse;
+  BuildingTenancy: string;
+  RentableArea: number;
+  BuildingOccupantTypeId: number;
+  BuildingOccupantType?: BuildingOccupantType;
+  LeaseExpiry?: Date;
+  OccupantName?: string;
+  TransferLeaseOnSale?: boolean;
+  BuildingTenancyUpdatedOn?: Date;
+  EncumbranceReason?: string;
+  LeasedLandMetadata?: string;
+  TotalArea?: number;
+  AdministrativeAreaId: number;
+  AdministrativeArea?: string;
+  IsVisibleToOtherAgencies?: boolean;
+  Location: GeoPoint;
+  Address1: string;
+  Address2: string;
+  ClassificationId: number;
+  Classification: Classification;
+  ProjectNumbers?: string;
+  PropertyTypeId: number;
+  PropertyType?: PropertyType;
+  AgencyId: number;
+  Agency?: Agency | null;
+  PID?: number;
+  PIN?: number;
+  IsSensitive: boolean;
+  Evaluations?: Evaluation[] | null;
+  Fiscals?: Fiscal[] | null;
+}
+
+const useBuildingsApi = (absoluteFetch: IFetch) => {
+  const addBuilding = async (building: Building) => {
```

Comment:
After fixing up the types/interfaces, consider using a more specify type for building. When adding a building, there are a number of properties it won't have yet (like ID), but this suggests it could be provided.